### ISSUE TITLE
Special commands do not work after the first command in a series

### ISSUE THREAD

**Issue by drocco007** [amjith/pgcli#160]
In pgcli and psql, this works:

    SELECT 1; SELECT 1;

However, using a special command after the first command only works in psql:

    psql (9.3.5)
    Type "help" for help.

    postgres=# select 1; \l
     ?column? 
    ----------
            1
    (1 row)

                                 List of databases
       Name    |  Owner   | Encoding | Collate | Ctype |   Access privileges   
    -----------+----------+----------+---------+-------+-----------------------
     _test_db  | postgres | UTF8     | C       | C     | 
     docker    | docker   | UTF8     | C       | C     | 
     foo       | postgres | UTF8     | C       | C     | 
     postgres  | postgres | UTF8     | C       | C     | 
     template0 | postgres | UTF8     | C       | C     | =c/postgres          +
               |          |          |         |       | postgres=CTc/postgres
     template1 | postgres | UTF8     | C       | C     | =c/postgres          +
               |          |          |         |       | postgres=CTc/postgres
    (6 rows)

    postgres=# 

while pgcli throws an error:

    postgres> SELECT 1; \l
    syntax error at or near "\"
    LINE 1: \l
            ^

    postgres>

**Comment by amjith** [comment]
You're right. This fails because the splitting of sql statements happens after an attempt is made to execute  special commands, whereas the splitting should really be happening before attempting any execution at all. 

Ref: https://github.com/amjith/pgcli/blob/master/pgcli/pgexecute.py#L141

I'll take a stab at it after I get some sleep. :)

**Comment by amjith** [comment]
It is now fixed in master. Can you give it a try and let me know if it works as expected? 

You can install directly from git using the following command: 

```
pip install -U -e git+https://github.com/amjith/pgcli.git#egg=pgcli
```

**Comment by drocco007** [comment]
Looks good!

After I posted the issue, I noticed that psql only allows special commands as the last command in the sequence. That is, this command

    \l ; SELECT 1;

now works in pgcli but produces warnings and does not execute the SELECT in psql. I probably wouldn't worry about it if it were me, but I thought I should mention it since I'm not sure how close to psql's behavior you're trying to tack.

Also, because of my change in #159, special commands aren't included in the suggestions if they are not the first command. Software is hard. :-/ This seems like a separate issue to me, so I'll be happy to open another issue for it if you'd like.

**Comment by amjith** [comment]
Yeah I knew about the psql limitations with multiple statements in the same line. But I'm sure it's a bug on their part and I decided not to copy that bug. 

I forgot about the completion of special command completion after a semi-colon. I've opened a new issue for that #164.

I probably won't get to it right away since it seems like a minor annoyance rather than a bug. 

Once again thanks for catching the corner cases. :)
